### PR TITLE
[minor] [build] Use custom temp directory during build.

### PR DESCRIPTION
**Pull request by vanzin**
Even with all the efforts to cleanup the temp directories created by
unit tests, Spark leaves a lot of garbage in /tmp after a test run.
This change overrides java.io.tmpdir to place those files under the
build directory instead.

After an sbt full unit test run, I was left with > 400 MB of temp
files. Since they're now under the build dir, it's much easier to
clean them up.

Also make a slight change to a unit test to make it not pollute the
source directory with test data.

### CODE REVIEW

**Comment by JoshRosen** [comment]
Nice catch. This plays nicely with some Jenkins tuning work that I'm doing, where we're experimenting with using ramdisks to store the Jenkins workspaces for the pull request builders.

File: pom.xml [hunk 1]
```diff
@@ -1255,6 +1255,7 @@
             </environmentVariables>
             <systemProperties>
               <java.awt.headless>true</java.awt.headless>
+              <java.io.tmpdir>${project.build.directory}/tmp</java.io.tmpdir>
```
Comment (harishreedharan) [inline pom.xml]: ${project.build.directory}/target/tmp ?

Comment (vanzin) [inline pom.xml]: `${project.build.directory}` is already the `target` directory.

Comment (harishreedharan) [inline pom.xml]: ah, ok!

**Comment by harishreedharan** [comment]
+1

**Comment by JoshRosen** [comment]
Update: ramdisk didn't make a difference, so we're not using it.

**Comment by srowen** [comment]
Looks OK to me.

**Comment by JoshRosen** [comment]
We should probably merge this into branch-1.4 and 1.3 as well, since those are the more active backport branches; I think that Shane will appreciate the reduction in `/tmp` garbage on the Jenkins boxes :)

**Comment by JoshRosen** [comment]
Looks like this may have broken a SparkSubmitSuite test: https://amplab.cs.berkeley.edu/jenkins/job/Spark-Master-Maven-pre-YARN/2654/

We may need to ensure that the temp dir. exists before we run tests or something like that; haven't had time to investigate.  Can someone look at this and either revert this patch or push a hotfix?

**Comment by vanzin** [comment]
Looking. Worked for me locally but may have been a test ordering thing.

**Comment by andrewor14** [comment]
I reverted this. Could we file a JIRA for this actually? This is a borderline minor change that would be good to track.

**Comment by pwendell** [comment]
@vanzin @srowen this broke the Spark 1.3 builds, so I revert it in the Spark 1.3 branch.

**Pull request closed without merging**